config.js

```diff
@@ -286,30 +286,35 @@ exports.updateFromArgv = function() {
     }
 
     if (argv.komi) {
-        if (argv.komi.includes(`auto`)) {
```

Comment:
Since we have this repetition of ranked/unranked/null for a bunch of arguments (and also blitz, live and corr), you could consider in general using a pattern of defining a function that would do the check to reduce trivial repetition:

```
function warn_for_auto_komi(rankedunranked) {
    let arg = 'komi' + rankedunranked;
    for (let e of argv[arg].split(",")) {
        if (e === "auto") {
            console.log("Warning: /--"+arg+" auto/ has been renamed to /--"+arg+" automatic/\n");
            // this argv may be undefined if not used by bot admin, so we skip a line here, not below
        }
    }
}
```

And then:

```
for (let rankedunranked of ['', 'ranked', 'unranked']) {
    warn_for_auto_komi(rankedunranked);
    // maybe other checks here
}
```

Reply:
thats exactly what i was doing currently, it is unfinished but you can have a look below @windo

Reply:
we are on the same wavelength it seems looo :+1:

Reply:
@windo 

after yesterday's review with you, and @Dorus and @roy7 's long review, i now can understand this code and yes its more efficient

Reply:
Why not `"a,b,c".split(',').includes("auto")`?

Reply:
i was about to use an array now @Dorus same as in #152 

good idea indeed :+1:

Reply:
just rewrote it entirely, closing this so that we can review again :100: